verification/curator-service/api/src/controllers/sources.ts

```diff
@@ -113,7 +113,10 @@ export default class SourcesController {
     private async updateAutomationScheduleAwsResources(
         source: SourceDocument,
     ): Promise<void> {
-        if (source.isModified('automation.schedule.awsScheduleExpression')) {
+        // Careful here, source.isModified('automation.schedule.awsScheduleExpression')
```

Comment:
That's odd. I wonder if `isDirectModified` is better, here? All good if this new logic works as well.

https://mongoosejs.com/docs/api/document.html#document_Document-isDirectModified

Reply:
I tried isDirectModified but it didn't trigger as expected :( pretty obscure tbh... the added tests and manual checks that I've done make me quite confident this is fine.